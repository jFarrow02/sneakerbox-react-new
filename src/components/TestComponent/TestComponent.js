import React from 'react';

const TestComponent = props=>{

    return(
        <div>
            This is a test: 
            <h2>{props.foo}</h2>
        </div>
    )
};

export default TestComponent;