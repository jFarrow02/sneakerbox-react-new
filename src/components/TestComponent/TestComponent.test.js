import React from 'react';
import TestComponent from './TestComponent';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({adapter: new Adapter()});
const wrapper = shallow(<TestComponent foo='bar'/>);

it('should match a snapshot', ()=>{
    const tree = renderer.create(<TestComponent/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render when passed no props', ()=>{
    expect(wrapper.exists()).toBeTruthy();
});

it('should render foo', ()=>{
    expect(wrapper.text()).toMatch(/bar/);
})