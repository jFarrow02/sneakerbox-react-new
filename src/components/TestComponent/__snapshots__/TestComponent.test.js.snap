// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<div>
  This is a test:
  <h2 />
</div>
`;
